### PR TITLE
Ssh username (fixes #44)

### LINKED ISSUE #44: Can't specify ssh_username via machine_options
Because of:

```ruby
          machine_spec.location = {
              'driver_url' => driver_url,
              'driver_version' => Chef::Provisioning::AWSDriver::VERSION,
              'allocated_at' => Time.now.utc.to_s,
              'host_node' => action_handler.host_node,
              'image_id' => machine_options[:image_id],
              'instance_id' => instance.id
          }
```

And:

```ruby
          username = machine_spec.location['ssh_username'] || default_ssh_username
```

We can't set ssh_username as a machine_option.

### PR DESCRIPTION
**Pull request by jkeiser**

### CODE REVIEW

**Comment by jkeiser** [comment]
This fixes #44 .

File: lib/chef/provisioning/aws_driver/driver.rb [hunk 1]
```diff
@@ -37,8 +37,14 @@ def self.from_url(driver_url, config)
 
     def initialize(driver_url, config)
       super
-      credentials = aws_credentials.default
-      @region = credentials[:region]
+
+      scheme, profile_name, region = driver_url.split(':')
```
Comment (afiune) [inline lib/chef/provisioning/aws_driver/driver.rb]: `schema` is never used.

**Comment by afiune** [comment]
Testing!

**Comment by jkeiser** [comment]
@afiune did this work for you?

**Comment by afiune** [comment]
Yes it does.. with a minor changes.. I will try to commit what we fixed this week! :smile:

**Comment by afiune** [comment]
YEAH @jkeiser !! I added some fixes!! :+1: to merge!!! This will let @schisamo move forward with our delivery cluster automation!! :smile:

File: lib/chef/provisioning/aws_driver/driver.rb [hunk 2]
```diff
@@ -263,6 +272,16 @@ def ready_machine(action_handler, machine_spec, machine_options)
 
     end
 
+    def connect_to_machine(name, chef_server = nil)
```
Comment (afiune) [inline lib/chef/provisioning/aws_driver/driver.rb]: Pretty dope! :smile_cat:

**Comment by afiune** [comment]
Anything else I can do to merge this? :smile: happy to help

**Pull request closed without merging**